backend/core/domains/payments/cache_service.py

```
@@ -630,7 +628,40 @@ def invalidate_payment_plan_caches(self, plan_id: int = None, event_id: int = No
         self._invalidate_cache_patterns(patterns_to_invalidate)
         logger.info(f"Invalidated payment plan caches for plan_id: {plan_id}")
     
-    def invalidate_payment_method_caches(self, method_id: int = None, user_id: int = None):
+    def invalidate_transaction_caches(self, transaction_id: int = None, payment_id: int = None, gateway_id: int = None):
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)